e2eshark/run.py

```diff
@@ -459,19 +459,17 @@ def runCodeGeneration(
 
 # ∣input−other∣ ≤ atol + rtol × ∣other∣
 # returns (atol, rtol)
-# Pytorch torch.testing defaults:
-#    bf16: atol=1e-05, rtol=1.6e-02
-#    fp16: atol=1e-05, rtol=1e-03
-#    fp32: atol=1e-05, rtol=1.3e-06
-def getTolerances(torchdtype):
-    if torchdtype == torch.bfloat16:
-        return (1e-02, 1e-01)
-    elif torchdtype == torch.float16:
-        return (1e-04, 1e-03)
+# Pytorch torch.testing:
+def getTolerances(args):
+    if args.atol and args.rtol:
+        return (float(args.atol), float(args.rtol))
+    elif args.atol:
+        return (float(args.atol), 1e-03)
+    elif args.rtol:
+        return (1e-03, float(args.rtol))
     else:
         return (1e-04, 1e-04)
```

Comment:
If these values aren't conditional on anything at runtime (like the data type or specific test case), you can just set default values on the argument parser instead of using a helper function.

```python
    parser.add_argument(
        "--rtol",
        help="Set rtol tolerance parameter",
        type=float,
        default=1e03,
    )
    parser.add_argument(
        "--atol",
        help="Set atol tolerance parameter",
        type=float,
        default=1e03,
    )

    inferencematched = torch.allclose(infooutput, goldoutput, args.rtol, args.atol)
```

Reply:
yes actually I did not even think about that ; thanks now it is much cleaner